### ISSUE TITLE
Feature request: Highlight all matches not just the one the cursor is at

### ISSUE THREAD

**Issue by tobimensch** [philc/vimium#2255]
When searching with I want to see all matches instantly (i.e. with yellow highlighting) without having to press n/N.

**Comment by marcelocra** [comment]
I'm **interested in working on this** and believe it **adheres to Vimium's design principles** perfectly.

Before I start, I want to know a couple of things. @philc , feel free to point me to any relevant docs or tag other maintainers that could answer these questions:

1. Is there any active effort in this direction? I don't want to duplicate work.
2. I've been using Vimium for a long time now and almost never use it for search, since <kbd>ctrl+f </kbd> does the job really well. The only problem is that it doesn't support regex. Perhaps this feature would be more interesting with regex? But I'm sure other people already thought of this and maybe it isn't feasible?
3. This would be my first contribution to Vimium. I looked at the [maintainer's guide](https://github.com/philc/vimium/wiki/Maintainer's-guide) and at the [CONTRIBUTING.md](./CONTRIBUTING.md) file and they don't mention anything about the architecture. I've worked on many chrome extensions and know what to look for, but still think such description could be useful there. Perhaps it already exists somewhere, but if it doesn't, I'm using Sonnet 4 to help me understand the codebase and could send it as a separate pull request.

<details>
<summary>Appendix: Vimium Architecture Summary</summary>

> **Appendix: Vimium Architecture Summary**
> 
> **tl;dr**
> 
> **Essential Starting Points for Contributors:**
> - Start with `manifest.json` for overall structure, then
> - `background_scripts/main.js` for background logic, then
> - `content_scripts/vimium_frontend.js` for content script coordination, and
> - `background_scripts/all_commands.js` for understanding available commands.
> 
> **Detailed Description**
> 
> **Core Entry Points:**
> - **Background Service Worker** ([background_scripts/main.js](https://github.com/philc/vimium/blob/master/background_scripts/main.js)): Extension initialization, tab management, and message routing between components
> - **Content Script Frontend** ([content_scripts/vimium_frontend.js](https://github.com/philc/vimium/blob/master/content_scripts/vimium_frontend.js)): Main content script entry point that coordinates all page functionality
> - **Content Scripts Aggregator** ([pages/all_content_scripts.js](https://github.com/philc/vimium/blob/master/pages/all_content_scripts.js)): Imports all content script modules for consistent loading
> 
> **Key Components:**
> - **Mode System** ([content_scripts/mode.js](https://github.com/philc/vimium/blob/master/content_scripts/mode.js)): Base class for managing different interaction modes
> - **Normal Mode** ([content_scripts/mode_normal.js](https://github.com/philc/vimium/blob/master/content_scripts/mode_normal.js)): Handles command processing and key mapping
> - **Vomnibar** ([content_scripts/vomnibar.js](https://github.com/philc/vimium/blob/master/content_scripts/vomnibar.js)): Omnisearch interface using iframe-based UI components
> - **Link Hints** ([content_scripts/link_hints.js](https://github.com/philc/vimium/blob/master/content_scripts/link_hints.js)): Keyboard-driven link navigation with visual markers
> - **Find Mode** ([content_scripts/mode_find.js](https://github.com/philc/vimium/blob/master/content_scripts/mode_find.js)): In-page search functionality
> - **Visual Mode** ([content_scripts/mode_visual.js](https://github.com/philc/vimium/blob/master/content_scripts/mode_visual.js)): Text selection mode
> - **Insert Mode** ([content_scripts/mode_insert.js](https://github.com/philc/vimium/blob/master/content_scripts/mode_insert.js)): Form input handling
> - **Marks System** ([content_scripts/marks.js](https://github.com/philc/vimium/blob/master/content_scripts/marks.js)): Local and global position bookmarks
> - **Scroller** ([content_scripts/scroller.js](https://github.com/philc/vimium/blob/master/content_scripts/scroller.js)): Smooth scrolling implementation
> - **HUD** ([content_scripts/hud.js](https://github.com/philc/vimium/blob/master/content_scripts/hud.js)): Heads-up display for status messages
> 
> **Background Services:**
> - **Completion Engine** ([background_scripts/completion.js](https://github.com/philc/vimium/blob/master/background_scripts/completion.js)): History, bookmarks, and search suggestions
> - **Tab Operations** ([background_scripts/tab_operations.js](https://github.com/philc/vimium/blob/master/background_scripts/tab_operations.js)): Tab management and navigation
> - **Command Registry** ([background_scripts/all_commands.js](https://github.com/philc/vimium/blob/master/background_scripts/all_commands.js)): Complete command definitions and metadata
> 
> **Core Libraries:**
> - **Settings Manager** ([lib/settings.js](https://github.com/philc/vimium/blob/master/lib/settings.js)): Configuration and preferences handling
> - **Handler Stack** ([lib/handler_stack.js](https://github.com/philc/vimium/blob/master/lib/handler_stack.js)): Event delegation and bubbling system
> - **DOM Utils** ([lib/dom_utils.js](https://github.com/philc/vimium/blob/master/lib/dom_utils.js)): DOM manipulation utilities
> - **Keyboard Utils** ([lib/keyboard_utils.js](https://github.com/philc/vimium/blob/master/lib/keyboard_utils.js)): Key event processing and mapping
> - **URL Utils** ([lib/url_utils.js](https://github.com/philc/vimium/blob/master/lib/url_utils.js)): URL validation and processing
> 
> **Build & Configuration:**
> - **Manifest Configuration** ([manifest.json](https://github.com/philc/vimium/blob/master/manifest.json)): Extension permissions and content script declarations
> - **Build System** ([make.js](https://github.com/philc/vimium/blob/master/make.js)): Deno-based build process with cross-browser manifest generation
> - **Test Stubs** ([lib/chrome_api_stubs.js](https://github.com/philc/vimium/blob/master/lib/chrome_api_stubs.js)): Chrome API mocks for testing environments
> 

</details>

### Improvement to Vimium Search

The feature should work exactly like <kbd>ctrl + f</kbd> in the browser, see image example:

1. Keep a persistent HTML element showing:
    1. the current search
    2. the number of matches
    3. the number of the current match
1. Show all matches in yellow
1. Show the current match in orange
1. Hide everything if the user presses <kbd>esc</kbd>
1. Allow the user to edit the current search if they press <kbd>/</kbd> when navigating the searches

<img width="744" height="340" alt="Image" src="https://github.com/user-attachments/assets/e55ff683-93cf-4b9f-baa8-73d055bbdfd4" />

</details>

**Comment by philc** [comment]
This UX has been long requested. The difficulty is that it's not easy for extensions to implement in a performant way, unless something has changed recently to make it more viable. @marcelocra if you'd like to do a survey of what's possible today and report back, that would be informative. Review the discussions in #591.
